fastplotlib/graphics/linecollection.py

```diff
@@ -49,33 +48,30 @@ def __init__(self, data: List[np.ndarray], z_position: Union[List[float], float]
             else:
                 _cmap = cmap
 
-            self.collection.append(LineGraphic(d, _z, _size, _colors, _cmap))
-
-    def _reset_feature(self):
-        pass
-
-    @property
-    def indices(self) -> Any:
-        pass
-
-    @indices.setter
-    @abstractmethod
-    def indices(self, indices: Any):
-        pass
+            self.data.append(LineGraphic(d, _z, _size, _colors, _cmap))
 
     @property
     def features(self) -> List[str]:
-        pass
+        return ["colors", "data"]
 
     def _set_feature(self, feature: str, new_data: Any, indices: Any):
-        if feature in ["colors", "data"]:
-            update_func = getattr(self, f"update_{feature}")
-            self.collection[indices].update_func(new_data)
+        if feature in self.features:
+            update_func = getattr(self.data[indices], f"update_{feature}")
```

Comment:
the way it's written it'll only work with single integer indices, we can chat about it

Reply:
once we implement this it'll be much easier https://github.com/kushalkolar/fastplotlib/issues/76#issuecomment-1358903126 

it would become `getattr(self, feature)[indices] = new_data`